lib/reservation_manager.rb

```diff
@@ -0,0 +1,136 @@
+require_relative "reservation.rb"
+require "date"
+
+require "awesome_print"
+
+module Hotel
+  class ReservationManager
+    #add constant for room numbers
+    MAX_ROOMS = 20
+    ROOM_RATE = 200
+
+    def initialize
+      @reservations = []
+      @block = []
+    end
+
+    def all_rooms
+      rooms = (1..MAX_ROOMS).to_a
+      return rooms
+    end
+
+    def make_reservation(check_in, check_out)
+      # unsure how to raise argument error when room is unavailable since room_number is assigned from avail_rooms array
+      if available_rooms(check_in, check_out).length == 0
+        raise ArgumentError, "There are no available rooms for this time period"
+      else
+        room_number = available_rooms(check_in, check_out).first
+        # reservation_rate = reservation_cost(check_in, check_out)
+
+        new_reservation = Hotel::Reservation.new(room_number,
+                                                 check_in,
+                                                 check_out)
+      end
+      @reservations << new_reservation
+
+      return @reservations
```

Comment:
In software, there's a little bit of a pattern that when you make something, you return that new thing, even if `@reservations` is being modified. That being said, that's more of a "pattern," and not a hard rule-- just wanted to call it out :)